sys/net/link_layer/ieee802154/submac.c

```diff
@@ -358,13 +358,22 @@ int ieee802154_set_phy_conf(ieee802154_submac_t *submac, uint16_t channel_num,
         return -ENETDOWN;
     }
 
-    int res = ieee802154_radio_config_phy(dev, &conf);
+    int res;
+    if ((res = ieee802154_radio_request_set_trx_state(dev, IEEE802154_TRX_STATE_TRX_OFF)) < 0) {
+        return res;
+    }
+
+    res = ieee802154_radio_config_phy(dev, &conf);
 
     if (res >= 0) {
         submac->channel_num = channel_num;
         submac->channel_page = channel_page;
         submac->tx_pow = tx_pow;
     }
+    while (ieee802154_radio_confirm_set_trx_state(dev) == -EAGAIN) {}
```

Comment:
IMO (but as outside of this PR) it would be nice to have this line as static inline functions (and same for every other confirm function that one might want to wait for). That would make it trivial to replace the busy waiting with blocking waiting for better power efficiency, as only a single point in the code would be need to handle this.

Reply:
yes, I agree